src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            using (JsonDocument employeesToSend = JsonDocument.Parse(jsonString))
+            {
+                // regular send:
+                Mailbox.SendAllEmployeesData(employeesToSend.RootElement);
+
+                // modified elements send:
+                JsonNode modifiableDocument = JsonElement.DeepCopy(employeesToSend);
+                var employees = modifiableDocument as JsonObject;
+                Assert.Equal(2, employees.Count());
+
+                employees.Add(EmployeesDatabase.GetNextEmployee());
+                Assert.Equal(3, employees.Count());
+
+                Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+            }
+        }
+
+        /// <summary>
+        /// Parsing right to JsonNode if user knows data will be modified
+        /// </summary>
+        [Fact]
+        public static void TestParsingToJsonNode()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+            employees.Add(EmployeesDatabase.GetNextEmployee());
+            Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+
+            Assert.Equal(2, employees.Count());
+            Assert.True(employees.ContainsProperty("employee1"));
+            Assert.True(employees.ContainsProperty("employee2"));
+
+            JsonObject employee2 = employees.GetObjectProperty("employee2");
+            Assert.IsType<JsonString>(employee2["name"]);
+            Assert.Equal("Zoe", (JsonString)employee2["name"]);
+            Assert.IsType<JsonString>(employee2["surname"]);
+            Assert.Equal("Coder", (JsonString)employee2["surname"]);
+            Assert.IsType<JsonNumber>(employee2["age"]);
+            Assert.Equal(24, (JsonNumber)employee2["age"]);
+        }
+
+        /// <summary>
+        /// Copying JsoneNode
+        /// </summary>
+        [Fact]
+        public static void TestCopyingJsonNode()
+        {
+            JsonObject employee = EmployeesDatabase.GetManager();
+            JsonNode employeeCopy = JsonNode.DeepCopy(employee);
+
+            static bool recursiveEquals(JsonNode left, JsonNode right)
+            {
```

Comment:
NIT: I'm not sure if this has made it to the guidelines yet or not, but I believe that when you are using local functions we usually add them at the very end of the method and we do it with a return statement before it, which clearly shows that the code is a local function. For void methods like this one, we would just add a `return;` and then your local funciton.

Reply:
We don't have such a guideline, and I'd disagree with it... particularly here it makes more sense to say "this is a named piece of work that I'm defining", and then later you use it.

It's the same as a lambda, really.

One guideline we do have (not written down) is that local methods should be named the same as non-local methods (PascalCased).  At least, it's what Immo and I suggested to the VS team.